src/main/java/mappers/firstvoices/CategoryMapper.java

```diff
@@ -3,63 +3,91 @@
  */
 package mappers.firstvoices;
 
-import mappers.propertyreaders.PropertyReader;
-import org.nuxeo.client.objects.Document;
-
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import mappers.propertyreaders.PropertyReader;
+import org.nuxeo.client.objects.Document;
+import org.nuxeo.client.objects.Documents;
 
 /**
  * @author dyona
  *
  */
-public class CategoryMapper extends RelatedPhraseMapper {
-	
-    @Override
-    protected String getCacheQuery() {
-    	// Include all categories in the Shared Data folder
-        return "SELECT * FROM FVCategory WHERE (fva:dialect IS NULL OR ecm:parentId='" + documents.get("Categories").getId()
-        		+ "') AND ecm:isTrashed = 0 AND ecm:isVersion = 0";
-    }
+public class CategoryMapper extends DictionaryCachedMapper {
 
-    private void updateMainDocumentReference(ArrayList<String> docIDList) {
-        documents.get("current").setPropertyValue(Properties.WORD_CATEGORIES, docIDList);
-    }
+  @Override
+  protected String getCacheQuery() {
 
-    public CategoryMapper() {
-        super("FVCategory", Columns.CATEGORIES);
-        parentKey = "Categories";
-        cacheProperty = Properties.TITLE;
-        propertyReaders.add(new PropertyReader(Properties.WORD_CATEGORIES, Columns.CATEGORIES));
+    Boolean useLocal = isLocalCategories();
+
+    /* For Local Dialect categories add running parameter localCategories */
+    Documents dialectCategoriesDirectory = client.operation("Repository.Query")
```

Comment:
Consider including a parameter of maxResults = 1 since you are only doing this to discover if a local category exists: https://explorer.nuxeo.com/nuxeo/site/distribution/server-10.10/viewOperation/Repository.Query